net/mwan3/files/etc/init.d/mwan3

```diff
@@ -11,21 +14,47 @@ boot() {
 	rc_procd start_service
 }
 
-# FIXME
-# fd 1000 is an inherited lock file descriptor for preventing concurrent
-# init script executions. Close it here to prevent the mwan3 daemon from
-# inheriting it further to avoid holding the lock indefinitely.
-
 reload_service() {
-	/usr/sbin/mwan3 restart 1000>&-
+	/usr/sbin/mwan3 restart
+}
+
+start_tracker() {
+	local enabled interface
+	interface=$1
+	config_get_bool enabled $interface 'enabled' '0'
+	[ $enabled -eq 0 ] && return
+
+	mkdir -p "$MWAN3TRACK_STATUS_DIR/$interface"
+	procd_open_instance "track_${1}"
+	procd_set_param command /usr/sbin/mwan3track $interface
+	procd_set_param respawn
+	procd_close_instance
 }
 
 start_service() {
-	/usr/sbin/mwan3 start 1000>&-
+
+	config_load mwan3
+	config_foreach start_tracker interface
+
+	/usr/sbin/mwan3 start_
+
+	procd_open_instance rtmon_ipv4
+	procd_set_param command /usr/sbin/mwan3rtmon ipv4
+	procd_set_param respawn
+	procd_close_instance
+
+	if command -v ip6tables > /dev/null; then
+		procd_open_instance rtmon_ipv6
+		procd_set_param command /usr/sbin/mwan3rtmon ipv6
+		procd_set_param respawn
+		procd_close_instance
+	fi
+
+
 }
 
 stop_service() {
-	/usr/sbin/mwan3 stop 1000>&-
+	/usr/sbin/mwan3 stop_
```

Comment:
If we now switch to procd, wouldn't it be better to integrate the mwan3 tool into the procd service call? We can work with EXTRA_COMMAND so that we can add the additional calls for ifup/ifdown/wrap and so on. If someone wants to use the tool further, so we are backward compatible, we can change the mwan3 tool as a transitional solution so that it serves as a wrapper for the procd.

Reply:
Yes, this is an ugly hack. I was trying to make a smaller change, but you are right, this should all be in /etc/init.d/mwan3 and /usr/sbin/mwan3 should just be a light wrapper for that. I will fix this.

Reply:
I moved most of the work to the init.d script. I suppose we could go the whole distance and move the rest of the code from the sbin script, to the init.d script, export everything we need with EXTRA_COMMAND and then symlink /usr/sbin/mwan3 to /etc/init.d/mwan3.